### PR TITLE
feat(B-9): Add VALID ADDITION LINES to specialist prompts (P0 Enhancement)

### PR DESCRIPTION
**Pull request by RC918**
## Summary

P0 Enhancement to address the inline comments weakness identified in comparative analysis with Gemini/Cursor. The root cause: LLM doesn't reliably follow the "addition-lines-only" constraint in prompts, causing inline comments to be downgraded to file-level comments when the LLM points to context lines.

This PR provides an explicit list of valid addition line numbers in the specialist prompt, similar to how PR #4199 provided "VALID FILES IN THIS PR" to prevent file path hallucination.

**Changes:**
- Add `_extract_addition_lines_from_diff()` to parse unified diff and extract valid line numbers (only lines starting with `+`)
- Add `_format_line_ranges()` to format line numbers compactly (e.g., `1-3, 5, 7-9`)
- Update `_build_user_prompt()` to include "VALID ADDITION LINES" section
- Update CRITICAL CONSTRAINTS to explicitly require LLM to use line numbers from the valid list

## Review & Testing Checklist for Human

- [ ] **Verify diff parsing correctness**: The `_extract_addition_lines_from_diff()` function parses unified diff format. Test with edge cases: new files (`/dev/null`), deleted files, renamed files, multiple hunks per file, binary files.
- [ ] **Verify line number accuracy**: The line tracking logic increments for `+` and ` ` lines but not `-` lines. Manually verify against a real diff that the extracted line numbers match actual file line numbers.
- [ ] **Test with production PR**: Run multi-specialist review on a real PR and verify the "VALID ADDITION LINES" section appears correctly in logs/telemetry.
- [ ] **Monitor inline comment success rate**: After deployment, compare inline comment success rate vs file-level fallback rate to measure improvement.

**Recommended test plan:**
1. Create a test PR with known addition lines
2. Trigger multi-specialist review
3. Verify the prompt includes correct valid addition lines
4. Check if LLM findings use line numbers from the valid list

### Notes

This PR does not include unit tests for the new diff parsing logic. Consider adding tests for `_extract_addition_lines_from_diff()` as a follow-up, especially for edge cases.

Blueprint Alignment: Section 7 (Parallel Collaboration) - improving specialist accuracy.

---
Link to Devin run: https://app.devin.ai/sessions/55832c623b294639871d23b39290eae5
Requested by: Ryan Chen (@RC918)

---

> [!NOTE]
> Introduces explicit "VALID ADDITION LINES" in specialist prompts to ensure inline comments target only diff-added lines.
> 
> - Adds `_extract_addition_lines_from_diff()` to parse unified diffs and collect per-file addition line numbers
> - Adds `_format_line_ranges()` to present line numbers compactly (e.g., `1-3, 5, 7-9`)
> - Updates `_build_user_prompt()` to include the valid files list plus a new "VALID ADDITION LINES" section and stricter constraints requiring those line numbers for inline comments
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fba87a5d6ff3a1f568197547122cbb66861a18af. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/review_context/multi_specialist_reviewer.py [hunk 1]
```diff
@@ -720,6 +720,78 @@ def _extract_files_from_diff(self, diff_content: str) -> List[str]:
...
+        Extract valid addition line numbers from a unified diff.
+
+        P0 Enhancement: Provide explicit list of valid addition lines to LLM
+        to solve the problem of LLM not reliably following "addition-lines-only"
+        constraint. This gives the LLM a concrete list of line numbers it can
+        use for inline comments.
+
+        Args:
+            diff_content: The unified diff content
+
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/multi_specialist_reviewer.py]: [ARCHITECTURE] The `_build_user_prompt` method is growing in complexity by adding detailed diff analysis logic directly into prompt construction. This creates tight coupling between prompt formatting and diff analysis logic.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

### File-Level Comments

*The following comments could not be posted inline:*

**`handoff/20250928/40_App/orchestrator/review_context/multi_specialist_reviewer.py`**
> \[ARCHITECTURE\] The `_extract_addition_lines_from_diff` method combines diff parsing logic with line number tracking, mixing low-level string processing with business logic about valid comment locations. This violates separation of concerns and makes the method harder to test and maintain.
